### PR TITLE
Polaris v12 border migration

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Part of #10519

### WHAT is this pull request doing?

- Migrates the CSS custom properties for `border` token groups
- Migrates the component props for `border`

This the pre-v12 migration step to create a safe CSS custom property mapping to the new border tokens.

<details>
<summary>RegExp token search</summary>
<pre><code>(?:--p-border-radius-1|--p-border-radius-1_5-experimental|--p-border-radius-2|--p-border-radius-3|--p-border-radius-4|--p-border-radius-5|--p-border-radius-6|--p-border-width-1|--p-border-width-1-experimental|--p-border-width-2|--p-border-width-2-experimental|--p-border-width-4)(?!\d)</code></pre>
</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by sophschneider — APPROVED** [review]
💫

File: polaris-react/src/components/RadioButton/RadioButton.scss [hunk 1]
```diff
@@ -165,7 +165,7 @@
   }
 
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-border-width-2));
+  @include focus-ring($border-width: var(--p-border-width-0165));
```
Comment (lgriffee) [inline polaris-react/src/components/RadioButton/RadioButton.scss:168]:
```suggestion
  @include focus-ring($border-width: var(--p-border-width-050));
```

`--p-border-width-2` should be replaced with `--p-border-width-050` here.

File: polaris-react/src/components/OptionList/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -60,7 +60,7 @@
   width: 100%;
   height: 100%;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-border-width-2));
+  @include focus-ring($border-width: var(--p-border-width-0165));
```
Comment (lgriffee) [inline polaris-react/src/components/OptionList/components/Checkbox/Checkbox.scss:63]:
```suggestion
  @include focus-ring($border-width: var(--p-border-width-050));
```

`--p-border-width-2` should be replaced with `--p-border-width-050` here.

File: polaris-react/src/components/Checkbox/Checkbox.scss [hunk 3]
```diff
@@ -142,13 +142,12 @@
   width: 100%;
   height: 100%;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
-  @include focus-ring($border-width: var(--p-border-width-2));
+  @include focus-ring($border-width: var(--p-border-width-0165));
```
Comment (lgriffee) [inline polaris-react/src/components/Checkbox/Checkbox.scss:145]:
```suggestion
  @include focus-ring($border-width: var(--p-border-width-050));
```

`--p-border-width-2` should be replaced with `--p-border-width-050` here.

**Comment by sam-b-rose** [comment]
Addressed incorrect manual migrations here https://github.com/Shopify/polaris/pull/10608

**Pull request merged**